brave/brave-core
# fix(wallet): Send V2 Checksum Errors

## Description 
Introduces the new `Checksum` warning and error handling.


Resolves <https://github.com/brave/brave-browser/issues/26957>

Security issue for the designs approved here https://github.com/brave/security/issues/1176

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### Testing Checksum Warning
1. Go to the `Send` page
2. Select a token to send and enter an amount
3. Past in an address with all lowercase characters (example: 0x7d66c9ddaed3115d93bd1790332f3cd06cf52b14)
4. You should see the `checksum` warning (yellow border), and the `Review order` button should still be `enabled`
5. Click on the `How can I solve it?` button and you should see the `checksum` modal.
6. Click on `Visit https://etherscan.io` and search for the address.
7. Copy the correct formatted address from `etherscan` and past it back in the `Send` page.
8. It should fix the `error`.

https://user-images.githubusercontent.com/40611140/216678032-ba777f20-4a24-44fe-875c-962a67da41de.mov

### Testing Checksum Warning
1. Go to the `Send` page
2. Select a token to send and enter an amount
3. Past in an address with one character lowercase that shouldn't be (example: 0x7d66c9ddAEd3115d93Bd1790332f3Cd06Cf52B14)
4. You should see the `checksum` error (red border), and the `Review order` button should still be `disabled`
5. Click on the `How can I solve it?` button and you should see the `checksum` modal.
6. Click on `Visit https://etherscan.io` and search for the address.
7. Copy the correct formatted address from `etherscan` and past it back in the `Send` page.
8. It should fix the `error`.

https://user-images.githubusercontent.com/40611140/216678084-54ae5f0b-16b1-4bd2-a042-041af161d147.mov
